lib/src/Rating/Rating.test.tsx

```diff
@@ -1,87 +1,236 @@
-        expect(mountedWrapper.find("svg").last().hasClass("custom-svg-star")).toBeTruthy();
-        // Hollow star
-        mountedWrapper.setProps({ useHollow: true });
-        expect(mountedWrapper.find("svg").first().hasClass("custom-svg-star-hollow")).toBeTruthy();
-        expect(mountedWrapper.find("svg").last().hasClass("custom-svg-star")).toBeTruthy();
+    it("Should pass any other native html prop", () => {
+        const id: string = "my-id";
+
+        act(() => {
+            render(<Rating {...{ ...props, id }} />, container);
+        });
+        expect(container).toBeDefined();
+
+        expect(container.firstElementChild);
+        expect(container.querySelector("input").id).toEqual(id);
     });
 
-    it("Should render with default colors if colors passed incorrectly", () => {
-        mountedWrapper.setProps({ initialValue: 3, colors: ["#0F0"] as any });
-        expect(mountedWrapper.find("svg").first().find(".star-fill").first().prop("fill")).toEqual(initialColors[0]);
-        expect(mountedWrapper.find("svg").last().find(".star-fill").first().prop("fill")).toEqual(initialColors[1]);
-        mountedWrapper.setProps({ useHollow: true });
-        expect(mountedWrapper.find("svg").first().find(".star-fill").first().prop("fill")).toEqual(initialColors[0]);
-        expect(mountedWrapper.find("svg").last().find(".star-fill").first().prop("fill")).toEqual(initialColors[1]);
+    it("Should contain 5 svgs", () => {
+        act(() => {
+            render(<Rating {...props} />, container);
+        });
+        expect(container.querySelector(".rating-icons").children.length).toEqual(5);
+
+        act(() => {
+            render(<Rating {...{ ...props, max: 2 }} />, container);
+        });
+        expect(container.querySelector(".rating-icons").children.length).toEqual(2);
     });
 
-    it("Should accept custom colors only when an array of two colors are passed", () => {
-        const colors: [string, string] = ["#0F0", "#F00"];
-        mountedWrapper.setProps({ initialValue: 3, colors });
-        expect(mountedWrapper.find("svg").first().find(".star-fill").first().prop("fill")).toEqual(colors[0]);
-        expect(mountedWrapper.find("svg").last().find(".star-fill").first().prop("fill")).toEqual(colors[1]);
-        mountedWrapper.setProps({ useHollow: true });
-        expect(mountedWrapper.find("svg").first().find(".star-fill").first().prop("fill")).toEqual(colors[0]);
-        expect(mountedWrapper.find("svg").last().find(".star-fill").first().prop("fill")).toEqual(colors[1]);
+    it("should not be clicked if props.readyOnly is true", () => {
+        const readOnly: boolean = true;
+        const onChange: jest.Mock = jest.fn();
+        act(() => {
+            render(<Rating {...{ ...props, readOnly }} onChange={onChange} />, container);
+        });
+        const inputElement: HTMLInputElement = container.querySelector(".rating input");
+        Simulate.change(inputElement, { clientX: 657 });
```

Comment:
It is recommended by React's team that every action you make during your test must be wrapped by an `act` statement.
```suggestion
        act(() => Simulate.change(inputElement, { clientX: 657 }));
```

Reply:
Done